### PR TITLE
Per channel search provider

### PR DESCRIPTION
**Pull request by dylanmccall**
This rewrites the user session end of Kolibri's search provider in C. For the most part, this program simply passes searches on to kolibri-daemon. The reason for this rewrite is to make use of `g_dbus_connection_register_subtree` so it is possible to search specific channels by using different objects exported by the search provider.

For example, to search all channels:

```
$ gdbus call --session -d org.learningequality.Kolibri.Devel.SearchProvider -o /org/learningequality/Kolibri/Devel/SearchProvider -m org.gnome.Shell.SearchProvider2.GetInitialResultSet '["test"]'
(['c/70a06bd6819056e4b55141720f4633cb?197934f144305350b5820c7c4dd8e194', 'c/1fc794c02ed65a3a9107b17e11640132?e409b964366a59219c148f2aaa741f43'],)
```

Or to search a specific channel:

```
$ gdbus call --session -d org.learningequality.Kolibri.Devel.SearchProvider -o /org/learningequality/Kolibri/Devel/SearchProvider/channel_e409b964366a59219c148f2aaa741f43 -m org.gnome.Shell.SearchProvider2.GetInitialResultSet '["test"]'
(['c/1fc794c02ed65a3a9107b17e11640132?e409b964366a59219c148f2aaa741f43'],)
```

This branch also includes changes to the Kolibri desktop application in order to support being launched with a particular channel ID through the `x-kolibri-dispatch` URI scheme.

### CODE REVIEW

File: src/kolibri_gnome/launcher/application.py [hunk 1]
```diff
@@ -27,12 +28,38 @@ def do_open(self, files, n_files, hint):
             self.handle_uri(uri)
 
     def handle_uri(self, uri):
-        valid_url_schemes = ("kolibri-channel", )
+        valid_url_schemes = ("kolibri-channel", 'x-kolibri-dispatch')
```
Comment (danigm) [inline src/kolibri_gnome/launcher/application.py:31]: This is not used now, maybe it's good to remove the variable.

File: src/kolibri_gnome_search_provider/kolibri-gnome-search-provider.c [hunk 2]
```diff
@@ -0,0 +1,632 @@
...
+  g_clear_pointer(&self->search_provider_skeleton, g_object_unref);
+  g_clear_pointer(&self->kolibri_daemon, g_object_unref);
+
+  G_OBJECT_CLASS(kolibri_gnome_search_provider_parent_class)->dispose(gobject);
+}
+
+static void
+kolibri_gnome_search_provider_finalize(GObject *gobject)
+{
+  // KolibriGnomeSearchProvider *self = KOLIBRI_GNOME_SEARCH_PROVIDER (gobject);
```
Comment (danigm) [inline src/kolibri_gnome_search_provider/kolibri-gnome-search-provider.c:85]: This commented line could be removed.

**Review by danigm — CHANGES_REQUESTED** [review]
This is not building for me using flatpak with a clean clone/build, I've created a PR that solves the build problem that I've found: https://github.com/learningequality/kolibri-installer-gnome/pull/35

Besides that, this is working correctly. I've left a few comments about the code, but looks pretty good.

**Comment by dylanmccall** [comment]
Thanks for the review! I pushed a change which should fix the build issue you were encountering, as well as those review comments. I'll see if I can split this up into more manageable commits as well :)

**Review by danigm — APPROVED** [review]
LGTM

**Pull request merged**